app-libs/stf/src/stf_sgx.rs

```diff
@@ -16,34 +16,56 @@
 */
 
 #[cfg(feature = "test")]
-use crate::test_genesis::test_genesis_setup;
+use crate::test_genesis::{test_genesis_endowees, test_genesis_setup};
 use crate::{
-	helpers::{enclave_signer_account, get_shard_vaults, shard_creation_info, shard_vault},
-	Stf, ENCLAVE_ACCOUNT_KEY,
+	helpers::{
+		enclave_signer_account, get_shard_vaults, shard_creation_info, shard_vault,
+		shielding_target_genesis_hash,
+	},
+	parentchain_mirror::ParentchainMirror,
+	Stf, TrustedCall, TrustedCallSigned, ENCLAVE_ACCOUNT_KEY,
 };
 use codec::{Decode, Encode};
 use frame_support::traits::{OnTimestampSet, OriginTrait, UnfilteredDispatchable};
+use ita_assets_map::{AssetId, AssetTranslation, FOREIGN_ASSETS, NATIVE_ASSETS};
+use ita_parentchain_specs::MinimalChainSpec;
 use ita_sgx_runtime::{
-	ParentchainInstanceIntegritee, ParentchainInstanceTargetA, ParentchainInstanceTargetB,
+	Assets, ParentchainInstanceIntegritee, ParentchainInstanceTargetA, ParentchainInstanceTargetB,
+	ShardManagement, ShardMode, System,
 };
 use itp_node_api::metadata::{provider::AccessNodeMetadata, NodeMetadataTrait};
+use itp_pallet_storage::{
+	AssetsPalletStorage, AssetsPalletStorageKeys, EnclaveBridgeStorage, EnclaveBridgeStorageKeys,
+	ForeignAssetsPalletStorage, ForeignAssetsPalletStorageKeys, SidechainPalletStorage,
+	SidechainPalletStorageKeys, SystemPalletStorage, SystemPalletStorageKeys,
+};
 use itp_sgx_externalities::SgxExternalitiesTrait;
 use itp_sgx_runtime_primitives::types::Moment;
 use itp_stf_interface::{
+	parentchain_mirror_prefix,
 	parentchain_pallet::ParentchainPalletInstancesInterface,
 	sudo_pallet::SudoPalletInterface,
 	system_pallet::{SystemPalletAccountInterface, SystemPalletEventInterface},
 	ExecuteCall, ExecuteGetter, InitState, ShardCreationInfo, ShardCreationQuery, ShardVaultQuery,
 	StateCallInterface, StateGetterInterface, UpdateState,
 };
-use itp_stf_primitives::{error::StfError, traits::TrustedCallVerification};
+use itp_stf_primitives::{
+	error::StfError,
+	traits::TrustedCallVerification,
+	types::{ShardIdentifier, Signature},
+};
 use itp_storage::storage_value_key;
-use itp_types::parentchain::{AccountId, Hash, ParentchainCall, ParentchainId};
-use itp_utils::stringify::account_id_to_string;
+use itp_types::parentchain::{AccountId, BlockNumber, Hash, Index, ParentchainCall, ParentchainId};
+use itp_utils::{hex::hex_encode, stringify::account_id_to_string};
 use log::*;
 use sp_runtime::traits::StaticLookup;
 use std::{fmt::Debug, format, prelude::v1::*, sync::Arc, vec};
 
+/// Maximum number of accounts to retire per block.
+/// chose conservatively as we don't limit execution time here but don't want to cause forks by all means.
+/// Moreover, this means a heavy burst for the parentchain too, so be gentle
+const MAX_ACCOUNT_RETIREMENTS_PER_BLOCK: usize = 10;
```

Comment:
Once you are retired, you  have plenty of time anyhow XD

Reply:
yep